src/components/utils/PreviewedLink.tsx

```diff
@@ -0,0 +1,149 @@
+import { Component, Show, JSX, createSignal, Switch, Match, createEffect, onMount } from 'solid-js';
+
+import SafeLink from '@/components/utils/SafeLink';
+import { isTwitterUrl, isYoutubeVideoUrl } from '@/utils/url';
+
+import '@/types/twitterWidget.d';
+
+type SafeLinkProps = {
+  class?: string;
+  href: string;
+  children?: JSX.Element;
+};
+
+type OgpContents = {
+  url: URL | null;
+  title: string;
+  description: string;
+  image: URL | null;
+};
+
+const twitterUrl = (urlString: string): string => {
+  try {
+    const url = new URL(urlString);
+    url.host = 'twitter.com';
+    return url.href;
+  } catch {
+    return '';
+  }
+};
+
+const youtubeUrl = (urlString: string): string => {
+  try {
+    const originalUrl = new URL(urlString);
+    const iframeUrl = new URL('https://www.youtube.com/embed/');
+    iframeUrl.pathname += originalUrl.searchParams.get('v');
+    iframeUrl.searchParams.set('origin', window.location.href);
```

Comment:
```suggestion
    iframeUrl.searchParams.set('origin', window.location.origin);
```

URL全体ではなく、`origin`を渡せば良いようです。